src/js/yeldaWebchatInjector.js

```diff
@@ -401,6 +401,19 @@ class YeldaChat {
       }
       return
     }
+
+    // Triggered from webchat to listen leave viewport event
+    if (event.data && event.data.event && event.data.event === 'listenLeaveViewport') {
```

Comment:
1. In this function, there are multiple checks on `event.data`. And in the end, if we don't have it, this function does not do anything.
So to avoid checkin it multiple times, I'd suggest adding a check at the very top:
```js
if (!event.data) {
  return
}
```

2. Do you remember why `event.data` can be either an object (`event.data.event`) or a string?
Isn't it possible to unify the format?

Reply:
Initially we does not need complex object sent as event so string type used like this
```
parent.postMessage('openChat', '*')
```
later we need complex object sent through event 
```
parent.postMessage({ event: 'openUrl', data: url })
```
If we unify the format now we need to change it on webchat side common to all the bots, but the chat plugin is versionised result is error when chat-plugin is not updated on client site